# Import Packages

In [1]:
import pandas as pd
import numpy as np
import random

# repeated printouts
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

# datetime 
from datetime import timedelta, date, datetime
from dateutil.relativedelta import relativedelta

# Load the raw data

We are only loading the excel sheets we will use for our research purposes. The sheets we need are `Demographics`, `Presenting Problems`, `Emergency Crisis Services`.

In [48]:
# Read selected excel sheets into pandas dataframes and save in a dictionary
file_path = '../input/Dartmouth Data Set.xlsx'
sheet_names = ['Demographics', 'Presenting Problems', 'Emergency Crisis Services', 'living Situation change', 'In-patient admissions']

# create an empty dictionary to store the sheet data
sheet_data = {}

# loop through each sheet name and read the corresponding sheet data into a pandas dataframe
for sheet_name in sheet_names:
    sheet_data[sheet_name] = pd.read_excel(file_path, sheet_name)

# Save the data in variables and inspect them

In [49]:
# Retrieve the dataframes from the dictionary and save them in variables
demographics = sheet_data['Demographics']
presenting_problems = sheet_data['Presenting Problems']
emergency_crisis_services = sheet_data['Emergency Crisis Services']
living_situation_change = sheet_data['living Situation change']
in_patient_admissions = sheet_data['In-patient admissions']

In [4]:
# Explore the demographics dataframe
demographics.head()
demographics.columns
demographics.info()

Local ID                             Region        County  \
0  8085297     California : CA START East Bay  Contra Costa   
1   440987                New York : Region 3      Saratoga   
2  8008815     California : CA START East Bay       Alameda   
3   220396                New York : Region 3        Albany   
4  6570649  California : CA START San Andreas    Santa Cruz   

     Start Coordinator Date Enrolled in START  \
0        Mary Akhuetie             2020-12-30   
1      Natalia Wolosen             2020-12-29   
2  Sharon Cyrus-Savary             2020-12-29   
3      Matthew Wolcott             2020-12-29   
4        Ilian Navarro             2020-12-29   

                                     Status Status Date  \
0                                    Active  2020-12-30   
1             Inactive (stable functioning)  2021-08-09   
2  Inactive (no longer requesting services)  2021-06-27   
3     Inactive (inappropriate for services)  2021-06-11   
4                                    Active  2020-12-29   

                       Time Enrolled in START  \
0  Business Hours (Monday - Friday 8am - 5pm)   
1  Business Hours (Monday - Friday 8am - 5pm)   
2  Business Hours (Monday - Friday 8am - 5pm)   
3  Business Hours (Monday - Friday 8am - 5pm)   
4  Business Hours (Monday - Friday 8am - 5pm)   

        Source of referral to START Suitability of enrollment in START  ...  \
0  Case Manager/Service Coordinator                        Appropriate  ...   
1  Case Manager/Service Coordinator                        Appropriate  ...   
2  Case Manager/Service Coordinator                        Appropriate  ...   
3  Case Manager/Service Coordinator                        Appropriate  ...   
4  Case Manager/Service Coordinator                        Appropriate  ...   

  State sponsored ICF/DD stay in past 1 - 5 years (at enrollment to START)  \
0                                                 No                         
1                                                 No                         
2                                                 No                         
3                                                 No                         
4                                                 No                         

  Jailed in the past year (at enrollment to START)  \
0                                               No   
1                                               No   
2                                               No   
3                                               No   
4                                               No   

  Jailed in the past 1 - 5 years (at enrollment to START)  \
0                                                 No        
1                                                 No        
2                                                 No        
3                                                 No        
4                                                 No        

  Employment Status at Enrollment Grade In School      Current IEP  \
0                         Student         Grade 7  Yes: 2020-11-16   
1                         Student        Grade 11               No   
2                         Student         Grade 8  Yes: 2020-01-01   
3                    Not employed             NaN              NaN   
4                             NaN             NaN              NaN   

  Current 504 Plan                                School Setting  \
0               No                Special school-out of district   
1               No                            Other: BOCES 4:1:2   
2               No                    Special school-in district   
3              NaN                                           NaN   
4              NaN  Other: COE Program Held at Cabrillo College    

     Custody Status Child Lives With  
0  Parental custody          Parents  
1  Parental custody          Parents  
2  Parental custody          Parents  
3               NaN              NaN  
4               N

Index(['Local ID', 'Region', 'County', 'Start Coordinator',
       'Date Enrolled in START', 'Status', 'Status Date',
       'Time Enrolled in START', 'Source of referral to START',
       'Suitability of enrollment in START',
       'Individual/Caregiver reliable access to technology',
       'Presenting problems at time of enrollment', 'Services at Enrollment',
       'Date of birth', 'Gender', 'Race', 'Ethnicity',
       'Level of Intellectual Disability',
       'Current living situation at enrollment to START',
       'More than one living situation over the last 5 years?',
       'More than one living situation over the last year?',
       'Psychiatric hospitalization in past year (at enrollment to START)',
       'Psychiatric hospitalizations in past 1 - 5 years (at enrollment to START)',
       'Emergency department visits in past year (at enrollment to START)',
       'Emergency department visits in past 1-5 years (at enrollment to START)',
       'Law enforcement encounters i

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4986 entries, 0 to 4985
Data columns (total 48 columns):
 #   Column                                                                                  Non-Null Count  Dtype         
---  ------                                                                                  --------------  -----         
 0   Local ID                                                                                4986 non-null   object        
 1   Region                                                                                  4986 non-null   object        
 2   County                                                                                  4986 non-null   object        
 3   Start Coordinator                                                                       4947 non-null   object        
 4   Date Enrolled in START                                                                  4986 non-null   datetime64[ns]
 5   Status              

In [5]:
# Explore the emergency crisis services dataframe
emergency_crisis_services.head()
emergency_crisis_services.columns
emergency_crisis_services.info()

Individual ID                                         Region  \
0      11153125  New Hampshire : Region 2 - Pathways/Claremont   
1      11153125  New Hampshire : Region 2 - Pathways/Claremont   
2      11153125  New Hampshire : Region 2 - Pathways/Claremont   
3      11158347  New Hampshire : Region 9 - Community Partners   
4      11158347  New Hampshire : Region 9 - Community Partners   

  Date/Time of Contact                 Source of Contact  \
0  2016-08-16 08:40:00  Case Manager/Service Coordinator   
1  2018-04-27 15:00:00  Case Manager/Service Coordinator   
2  2018-06-05 16:00:00  Residential provider - Community   
3  2015-06-01 14:00:00                     Family member   
4  2015-06-01 16:00:00                     Family member   

                                  Reason for Contact  \
0  At risk of losing placement, Other, Self-injur...   
1             Mental health symptoms, Self-injurious   
2  Mental health symptoms, Suicidal ideation/beha...   
3  Diagnosis and treatment plan assistance, Other...   
4  Aggression (physical, verbal, property destruc...   

  Type of Emergency Response  \
0    Phone consultation only   
1  In-person: Emergency room   
2     Other: phone and email   
3    Phone consultation only   
4    Phone consultation only   

   Total episode length (initial call to disposition)  \
0                                               15.0    
1                                               60.0    
2                                              120.0    
3                                              254.0    
4                                               70.0    

   Total response time (initial call to arrival on-site)  Travel Time  \
0                                                NaN              NaN   
1                                              240.0             60.0   
2                                                NaN              NaN   
3                                                NaN              NaN   
4                                                NaN              NaN   

  Police Response  ... Restraints Used If Restraints Were Used  \
0              No  ...              No                     NaN   
1              No  ...              No                     NaN   
2              No  ...              No                     NaN   
3              No  ...              No                     NaN   
4              No  ...              No                     NaN   

                                 Disposition  \
0                  Other: outpatient MH appt   
1  START emergency resource center admission   
2             Psychiatric Hospital Admission   
3  START emergency resource center admission   
4                   Maintain current setting   

            Disposition Information Admitted through ED/CPEP  \
0                               NaN                       No   
1                               NaN                       No   
2                               NaN                       No   
3                               NaN                       No   
4  Planned Resource Center referral                       No   

  Date of Admission Disposition Information (ER)  \
0               NaT                          NaN   
1               NaT                          NaN   
2               NaT                          NaN   
3               NaT                          NaN   
4               NaT                          NaN   

  Did START response prevent higher level of care (hospital diversion)  \
0                                                 No                     
1                                                Yes                     
2                                                Yes                     
3                                                 No                     
4                                                 No                     

  RSQ Completed?         Entered By  
0             No  Valarie Tetreault  
1             No   Samantha

Index(['Individual ID', 'Region', 'Date/Time of Contact', 'Source of Contact',
       'Reason for Contact', 'Type of Emergency Response',
       'Total episode length (initial call to disposition)',
       'Total response time (initial call to arrival on-site)', 'Travel Time',
       'Police Response', 'If Police Responded', 'Who Contacted Police?',
       'Mobile Crisis', 'If Mobile Crisis', 'Restraints Used',
       'If Restraints Were Used', 'Disposition', 'Disposition Information',
       'Admitted through ED/CPEP', 'Date of Admission',
       'Disposition Information (ER)',
       'Did START response prevent higher level of care (hospital diversion)',
       'RSQ Completed?', 'Entered By'],
      dtype='object')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18168 entries, 0 to 18167
Data columns (total 24 columns):
 #   Column                                                                Non-Null Count  Dtype         
---  ------                                                                --------------  -----         
 0   Individual ID                                                         18168 non-null  object        
 1   Region                                                                18168 non-null  object        
 2   Date/Time of Contact                                                  18168 non-null  datetime64[ns]
 3   Source of Contact                                                     17780 non-null  object        
 4   Reason for Contact                                                    17983 non-null  object        
 5   Type of Emergency Response                                            17930 non-null  object        
 6   Total episode length (initial call to 

In [6]:
# Explore the presenting problems dataframe
presenting_problems.head()
presenting_problems.columns
presenting_problems.info()

Client Local Id START Coordinator                         Region  \
0               1     Ashley Gruver  Iowa : County Social Services   
1               1     Ashley Gruver  Iowa : County Social Services   
2               1     Ashley Gruver  Iowa : County Social Services   
3         000083W          Amy Huff          North Carolina : West   
4         000083W          Amy Huff          North Carolina : West   

                   Presenting Problems at Enrollment  
0  Aggression (physical, verbal, property destruc...  
1                             Mental health symptoms  
2                                    Self-injurious   
3  Aggression (physical, verbal, property destruc...  
4  Decrease in ability to participate in daily fu...

Index(['Client Local Id', 'START Coordinator', 'Region',
       'Presenting Problems at Enrollment'],
      dtype='object')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31014 entries, 0 to 31013
Data columns (total 4 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   Client Local Id                    31014 non-null  object
 1   START Coordinator                  29435 non-null  object
 2   Region                             31014 non-null  object
 3   Presenting Problems at Enrollment  31014 non-null  object
dtypes: object(4)
memory usage: 969.3+ KB


Let's skim `Presenting Problems at Enrollment` to see the presented problems

In [7]:
random.choices(presenting_problems['Presenting Problems at Enrollment'].unique(), k = 20)

['Other: Trauma, family dynamics',
 'Other: grief and loss, low sodium levels',
 'Other: transition from placement back home',
 'Other: Transition from Murdoch Developmental Center in next two weeks.',
 'Other: leaves/runs, is new to area',
 'Other: School issues, lying, stealing',
 'Other: J will take items that do not belong to her. J also consumes her own fecal matter',
 'Other: Walking away',
 'Other: Purposeful Ingestion and eloping',
 'Other: Recent TRAUMA',
 'Other: Social isolation',
 'Other: Medication refusal',
 'Other: Purposeful Ingestion and eloping',
 'Other: Pain Medication addiction, excessive marijuana use',
 'Other: Resource Center',
 'Other: Transition/support planning',
 'Other: Regressed social interactions',
 'Other: impulsivity and impaired judgement',
 'Other: residential transition',
 'Other: open wounds due to self-injury']

<font color = "green"> As we can see above the presented problems are all over the place. In the next section, we will focus on cleaning this column using regex to narrow down the presented problems.  </font>

# Data cleaning/preprocessing

Let's first `Presenting Problems at Enrollment` column to `presented_problems` for more readability.

In [8]:
# rename the column
presenting_problems.rename(columns={'Presenting Problems at Enrollment': 'presented_problems'}, inplace=True)

# confirm that the column was renamed successfully
presenting_problems.columns

Index(['Client Local Id', 'START Coordinator', 'Region', 'presented_problems'], dtype='object')

In [9]:
presenting_problems['presented_problems'].value_counts()

Aggression (physical, verbal, property destruction, threats)                                                                                        7427
Family needs assistance                                                                                                                             4489
Mental health symptoms                                                                                                                              4356
Self-injurious                                                                                                                                      2640
Diagnosis and treatment plan assistance                                                                                                             2246
                                                                                                                                                    ... 
Other: Social Skills                                                              

Now, let's turn all the presenting problems in lowercase to see add consistency.

In [10]:
presenting_problems['presented_problems'] = presenting_problems['presented_problems'].str.lower()
presenting_problems.sample(2)

Client Local Id START Coordinator                          Region  \
24364         547058C      Ashley Aipia           North Carolina : East   
24149         5243167      Yakira Evans  California : CA START East Bay   

             presented_problems  
24364  transition from hospital  
24149  transition from hospital

In [11]:
# subset to rows that do not contain 'other:' for presented problems
presenting_problems_no_other = presenting_problems[~presenting_problems["presented_problems"].str.contains('other:')].copy()
presenting_problems_no_other.presented_problems.unique()

presenting_problems_no_other.shape[0]

print(f"The number of unique presenting problems before dropping 'Other' is: {presenting_problems.presented_problems.nunique()}")
print(f"The number of unique presenting problems is: {presenting_problems_no_other.presented_problems.nunique()}")

array(['aggression (physical, verbal, property destruction, threats)',
       'mental health symptoms', 'self-injurious ',
       'decrease in ability to participate in daily functions',
       'leaving unexpectedly', 'diagnosis and treatment plan assistance',
       'family needs assistance', 'at risk of losing placement',
       'sexualized behavior', 'suicidal ideation',
       'transition from hospital', 'suicidal action'], dtype=object)

30011

The number of unique presenting problems before dropping 'Other' is: 704
The number of unique presenting problems is: 12


Let's see how many rows we are losing by removing presented problems that start with `Other:`

In [12]:
print(f"Retention Rate After Dropping 'Other': {round(100*presenting_problems_no_other.shape[0]/presenting_problems.shape[0], 2)}%")

Retention Rate After Dropping 'Other': 96.77%


A retention rate of 96.77% is really go, so we will stick with the 12 unique presented problems. Now, we will write a function to parse the `Region` variable to `state` and `region`

In [13]:
def parse_region(row):
    """
    Parse a string in the format "<state>: <region>" and return a tuple
    containing the state and region.
    
    Parameters:
    row (str): A string in the format "<state> : <region>".
    
    Returns:
    A tuple containing the parsed state and region.
    """
    state, region = row.lower().split(':')
    return state.strip(), region.strip()

# apply the function on the demographics, presenting_problems_no_other, and emergency_crisis_services dataframes
presenting_problems_no_other.loc[:, ['state', 'region_new']] = presenting_problems_no_other['Region'].apply(parse_region).tolist()
demographics.loc[:, ['state', 'region_new']] = demographics['Region'].apply(parse_region).tolist()
emergency_crisis_services.loc[:, ['state', 'region_new']] = emergency_crisis_services['Region'].apply(parse_region).tolist()

# sample from the presenting_problems_no_other dataframe
presenting_problems_no_other.sample(3)

Client Local Id START Coordinator                 Region  \
21892          435359     Athena Ackley    New York : Region 1   
23016         456226W     Sulina George  North Carolina : West   
17453          376560  Jennifer Siemann    New York : Region 5   

                            presented_problems           state region_new  
21892  diagnosis and treatment plan assistance        new york   region 1  
23016                  family needs assistance  north carolina       west  
17453                          self-injurious         new york   region 5

In [14]:
# check all the unique states in the presenting_problems_no_other dataframe
presenting_problems_no_other.state.unique()

array(['iowa', 'north carolina', 'texas', 'new york', 'california',
       'colorado', 'new hampshire', 'maryland'], dtype=object)

Next, we will merge the `demographic` and `presenting_problems_no_other` dataframes using the `Client ID`.

In [15]:
demographics_problems_merged = presenting_problems_no_other.merge(demographics, left_on='Client Local Id', right_on='Local ID',
                                                                  how='inner', suffixes=('_presenting_problems', '_demographics'), 
                                                                  indicator = True).drop_duplicates()

In [16]:
demographics_problems_merged.sample(3)
demographics_problems_merged.shape

Client Local Id START Coordinator  \
2108        11166977       Emily Hatch   
5499          256287     Ebony Johnson   
7206          315730      Dayna Passer   

                         Region_presenting_problems  \
2108  New Hampshire : Region 2 - Pathways/Claremont   
5499  New York : Region 4 Richmond Kings - Brooklyn   
7206                            New York : Region 1   

                                     presented_problems  \
2108            diagnosis and treatment plan assistance   
5499                             mental health symptoms   
7206  decrease in ability to participate in daily fu...   

     state_presenting_problems      region_new_presenting_problems  Local ID  \
2108             new hampshire       region 2 - pathways/claremont  11166977   
5499                  new york  region 4 richmond kings - brooklyn    256287   
7206                  new york                            region 1    315730   

                                Region_demographics    County  \
2108  New Hampshire : Region 2 - Pathways/Claremont  Sullivan   
5499  New York : Region 4 Richmond Kings - Brooklyn     Kings   
7206                            New York : Region 1      Erie   

     Start Coordinator  ... Employment Status at Enrollment Grade In School  \
2108       Emily Hatch  ...                    Not employed             NaN   
5499     Ebony Johnson  ...                    Not employed       Graduated   
7206      Dayna Passer  ...                 Employment (PT)   Not in school   

     Current IEP Current 504 Plan School Setting    Custody Status  \
2108          No               No            NaN               NaN   
5499         NaN              NaN            NaN  Parental custody   
7206         NaN              NaN            NaN               NaN   

     Child Lives With state_demographics             region_new_demographics  \
2108              NaN      new hampshire       region 2 - pathways/claremont   
5499          Parents           new york  region 4 richmond kings - brooklyn   
7206              NaN           new york                            region 1   

     _merge  
2108   both  
5499   both  
7206   both  

[3 rows x 57 columns]

(17808, 57)

Using the `Date Enrolled in START` variable in the merged data frame, we will make a binary classifier of whether or not the client enrolled pre or post COVID. We will use March 01, 2020 as the cutoff date.

In [17]:
# convert to datetime format and save in enrollment_date
demographics_problems_merged['enrollment_date'] = pd.to_datetime(demographics_problems_merged['Date Enrolled in START'])
print(f"Data type for the date variable: {demographics_problems_merged['enrollment_date'].dtype}")

# create a binary clarity column based on pre and post covid.
date_cutoff = pd.to_datetime("2020-03-01")
demographics_problems_merged['is_post_covid'] = demographics_problems_merged['enrollment_date'] >= date_cutoff

# check the value counts to see how many of the inputs are post covid
demographics_problems_merged['is_post_covid'].value_counts()

Data type for the date variable: datetime64[ns]


False    14270
True      3538
Name: is_post_covid, dtype: int64

In [18]:
# Check the range of the data frame
demographics_problems_merged['enrollment_date'].max()
demographics_problems_merged['enrollment_date'].min()

Timestamp('2020-12-30 00:00:00')

Timestamp('2009-09-15 00:00:00')

In [19]:
# creating enrollment year varibale to see cases by year
demographics_problems_merged['enrollment_year'] = pd.DatetimeIndex(demographics_problems_merged['enrollment_date']).year

The initial classifier seems to have way more data for the pre-covid data. So for a reasonable comparison, we will subset the data frame to an equal number of month pre and post covid start date.

In [20]:
# Create the lower and upper bound of the dataset
def compare_disparities(df: pd.DataFrame, month_bandwidth: int):
    pre_change_lower_bound = date_cutoff - relativedelta(months = month_bandwidth)
    post_change_upper_bound = date_cutoff + relativedelta(months = month_bandwidth)
    print(f"Pre Change Lower Bound: {pre_change_lower_bound}, Post Change Upper Bound: {post_change_upper_bound}")

    # Subset to data that is within the range of the pre and post covid data
    filtered_combined = df[(df['enrollment_date'] >= pre_change_lower_bound) &
                                                    (df['enrollment_date'] <= post_change_upper_bound)].copy()

    print(f"The shape for the filtered data frame is {filtered_combined.shape}")

    filtered_combined['is_post_covid'] = df['enrollment_date'] >= date_cutoff

    return filtered_combined

# Preliminary data analysis

In [21]:
# check the value counts to see how many of the inputs are post covid
filtered_combined_10 = compare_disparities(demographics_problems_merged, 10)
filtered_combined_10['is_post_covid'].value_counts()

Pre Change Lower Bound: 2019-05-01 00:00:00, Post Change Upper Bound: 2021-01-01 00:00:00
The shape for the filtered data frame is (7697, 60)


False    4159
True     3538
Name: is_post_covid, dtype: int64

In [22]:
def count_distribution(combined: pd.DataFrame):
    # see how the frequency of presented problem changed pre and post covid
    df = pd.DataFrame(combined.groupby('is_post_covid').agg({'presented_problems' : 'value_counts'}).
                    rename(columns={'presented_problems' : 'presented_problems_count'}))

    # pivot the table to get the counts of presented problems pre and post covid
    return pd.pivot_table(df, index='is_post_covid', columns='presented_problems')

count_distribution(filtered_combined_10)

print()
# since the number entries for each presented problem before and post covid is different, 
# we need to normalize the value counts
def count_proportion(combined: pd.DataFrame):
    df = pd.DataFrame(combined.groupby('is_post_covid').agg({'presented_problems' : 'value_counts'}).
                rename(columns={'presented_problems' : 'presented_problems_count'}))
    
    df['presented_problems_count'] = (df['presented_problems_count']/df['presented_problems_count'].sum())*100

    return pd.pivot_table(df, index='is_post_covid', columns='presented_problems')

count_proportion(filtered_combined_10)

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
is_post_covid                                                                     
False                                                             975             
True                                                              796             

                                                \
presented_problems at risk of losing placement   
is_post_covid                                    
False                                      235   
True                                       176   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
is_post_covid                                                              
False                                                             337      
True                                                              258      

                                                            \
presented_problems diagnosis and treatment plan assistance   
is_post_covid                                                
False                                                  309   
True                                                   289   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
is_post_covid                                                     
False                                  640                  296   
True                                   575                  245   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
is_post_covid                                                                   
False                                 568             367                 163   
True                                  502             325                 122   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
is_post_covid                                                                  
False                            2               183                       84  
True                             9               156                       85

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
is_post_covid                                                                     
False                                                       12.667273             
True                                                        10.341692             

                                                \
presented_problems at risk of losing placement   
is_post_covid                                    
False                                 3.053138   
True                                  2.286605   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
is_post_covid                                                              
False                                                        4.378329      
True                                                         3.351955      

                                                            \
presented_problems diagnosis and treatment plan assistance   
is_post_covid                                                
False                                             4.014551   
True                                              3.754710   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
is_post_covid                                                     
False                             8.314928             3.845654   
True                              7.470443             3.183058   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
is_post_covid                                                                   
False                            7.379499        4.768091            2.117708   
True                             6.522022        4.222424            1.585033   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
is_post_covid                                                                  
False                     0.025984          2.377550                 1.091334  
True                      0.116929          2.026764                 1.104326

In [23]:
# see all the unique presented problems 
filtered_combined_10.presented_problems.unique()

array(['aggression (physical, verbal, property destruction, threats)',
       'decrease in ability to participate in daily functions',
       'at risk of losing placement', 'family needs assistance',
       'diagnosis and treatment plan assistance',
       'mental health symptoms', 'self-injurious ',
       'leaving unexpectedly', 'sexualized behavior', 'suicidal ideation',
       'transition from hospital', 'suicidal action'], dtype=object)

In [24]:
presenting_problems.Region.value_counts()

New York : Region 1                                            4132
New York : Region 3                                            3658
Texas : Tarrant County                                         2445
North Carolina : Central                                       2241
New York : Region 5                                            1933
North Carolina : West                                          1830
New York : Region 4 Richmond Kings - Brooklyn                  1351
North Carolina : East                                          1341
California : CA START San Diego                                 905
New York : Region 4 Tri-Borough - Queens                        885
California : CA START San Andreas                               881
New Hampshire : Region 4 - Community Bridges                    847
New York : Region 4 Tri-Borough - The Bronx                     791
Texas : El Paso County                                          784
Iowa : County Social Services                   

In [25]:
# creating new year variable to use groupby
filtered_combined_10['enrollment_year'] = pd.DatetimeIndex(filtered_combined_10['enrollment_date']).year
filtered_combined_10['enrollment_year'].value_counts()

# function to find distribution of problems by year
def count_year_distribution(combined: pd.DataFrame):
    # see how number of presented problems changed per year
    df = pd.DataFrame(combined.groupby('enrollment_year').agg({'presented_problems' : 'value_counts'}).
                    rename(columns={'presented_problems' : 'presented_problems_count'}))

    # pivot the table to get the counts of presented problems per year
    return pd.pivot_table(df, index='enrollment_year', columns='presented_problems')

count_year_distribution(filtered_combined_10)

# function to find proportion of problems by year
def count_year_proportion(combined: pd.DataFrame):
    # see how proportion of presented problems changed per year
    df = pd.DataFrame(combined.groupby('enrollment_year').agg({'presented_problems' : 'value_counts'}).
                rename(columns={'presented_problems' : 'presented_problems_count'}))
    
    df['presented_problems_count'] = (df['presented_problems_count']/df['presented_problems_count'].sum())*100

    # pivot the table to get the proportions of presented problems per year
    return pd.pivot_table(df, index='enrollment_year', columns='presented_problems')

count_year_proportion(filtered_combined_10)

2020    4373
2019    3324
Name: enrollment_year, dtype: int64

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
enrollment_year                                                                   
2019                                                              788             
2020                                                              983             

                                                \
presented_problems at risk of losing placement   
enrollment_year                                  
2019                                       198   
2020                                       213   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
enrollment_year                                                            
2019                                                              262      
2020                                                              333      

                                                            \
presented_problems diagnosis and treatment plan assistance   
enrollment_year                                              
2019                                                   247   
2020                                                   351   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
enrollment_year                                                   
2019                                   515                  239   
2020                                   700                  302   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
enrollment_year                                                                 
2019                                  444             290                 127   
2020                                  626             402                 158   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
enrollment_year                                                                
2019                             1               145                       68  
2020                            10               194                      101

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
enrollment_year                                                                   
2019                                                        10.237755             
2020                                                        12.771210             

                                                \
presented_problems at risk of losing placement   
enrollment_year                                  
2019                                  2.572431   
2020                                  2.767312   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
enrollment_year                                                            
2019                                                         3.403924      
2020                                                         4.326361      

                                                            \
presented_problems diagnosis and treatment plan assistance   
enrollment_year                                              
2019                                              3.209042   
2020                                              4.560218   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
enrollment_year                                                   
2019                              6.690919             3.105106   
2020                              9.094452             3.923607   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
enrollment_year                                                                 
2019                             5.768481        3.767702            1.649994   
2020                             8.133039        5.222814            2.052748   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
enrollment_year                                                                
2019                      0.012992          1.883851                 0.883461  
2020                      0.129921          2.520463                 1.312200

In [26]:
# checking number of data in each
filtered_combined_10.enrollment_year.value_counts()

2020    4373
2019    3324
Name: enrollment_year, dtype: int64

In [27]:
# filtering by 3 months before and after to see immediate impact
filtered_combined_3 = compare_disparities(demographics_problems_merged, 3)
filtered_combined_3['is_post_covid'].value_counts()

Pre Change Lower Bound: 2019-12-01 00:00:00, Post Change Upper Bound: 2020-06-01 00:00:00
The shape for the filtered data frame is (2313, 60)


False    1237
True     1076
Name: is_post_covid, dtype: int64

In [28]:
# running analysis based on 3 months before and after
count_distribution(filtered_combined_3)
count_proportion(filtered_combined_3)

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
is_post_covid                                                                     
False                                                           278.0             
True                                                            257.0             

                                                \
presented_problems at risk of losing placement   
is_post_covid                                    
False                                     61.0   
True                                      53.0   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
is_post_covid                                                              
False                                                           104.0      
True                                                             83.0      

                                                            \
presented_problems diagnosis and treatment plan assistance   
is_post_covid                                                
False                                                 99.0   
True                                                  85.0   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
is_post_covid                                                     
False                                182.0                 81.0   
True                                 170.0                 79.0   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
is_post_covid                                                                   
False                               184.0           115.0                50.0   
True                                160.0            98.0                32.0   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
is_post_covid                                                                  
False                          1.0              60.0                     22.0  
True                           NaN              41.0                     18.0

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
is_post_covid                                                                     
False                                                       12.019023             
True                                                        11.111111             

                                                \
presented_problems at risk of losing placement   
is_post_covid                                    
False                                 2.637268   
True                                  2.291396   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
is_post_covid                                                              
False                                                        4.496325      
True                                                         3.588413      

                                                            \
presented_problems diagnosis and treatment plan assistance   
is_post_covid                                                
False                                             4.280156   
True                                              3.674881   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
is_post_covid                                                     
False                             7.868569             3.501946   
True                              7.349762             3.415478   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
is_post_covid                                                                   
False                            7.955037        4.971898            2.161695   
True                             6.917423        4.236922            1.383485   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
is_post_covid                                                                  
False                     0.043234          2.594034                 0.951146  
True                           NaN          1.772590                 0.778210

In [29]:
filtered_combined_3.presented_problems.unique()

array(['aggression (physical, verbal, property destruction, threats)',
       'decrease in ability to participate in daily functions',
       'family needs assistance', 'leaving unexpectedly',
       'mental health symptoms', 'self-injurious ',
       'diagnosis and treatment plan assistance', 'suicidal ideation',
       'at risk of losing placement', 'sexualized behavior',
       'transition from hospital', 'suicidal action'], dtype=object)

In [30]:
# count presenting problems by year
count_year_distribution(demographics_problems_merged)
count_year_proportion(demographics_problems_merged)

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
enrollment_year                                                                   
2009                                                              1.0             
2010                                                              5.0             
2011                                                             22.0             
2012                                                             16.0             
2013                                                             38.0             
2014                                                            175.0             
2015                                                            143.0             
2016                                                            243.0             
2017                                                            578.0             
2018                                                            922.0             
2019                                                           1136.0             
2020                                                            983.0             

                                                \
presented_problems at risk of losing placement   
enrollment_year                                  
2009                                       NaN   
2010                                       3.0   
2011                                      15.0   
2012                                      10.0   
2013                                      19.0   
2014                                      39.0   
2015                                      39.0   
2016                                      57.0   
2017                                     150.0   
2018                                     187.0   
2019                                     267.0   
2020                                     213.0   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
enrollment_year                                                            
2009                                                              1.0      
2010                                                              2.0      
2011                                                              7.0      
2012                                                             10.0      
2013                                                             17.0      
2014                                                             32.0      
2015                                                             33.0      
2016                                                             53.0      
2017                                                            164.0      
2018                                                            300.0      
2019                                                            392.0      
2020                                                            333.0      

                                                            \
presented_problems diagnosis and treatment plan assistance   
enrollment_year                                              
2009                                                   1.0   
2010                                                   5.0   
2011                                                  14.0   
2012                                                  16.0   
2013                                                  21.0   
2014                                                  33.0   
2015                                                  32.0   
2016                                                  41.0   
2017                                                 158.0   
2018                                                 293.0   
2019                                                 353.0   
2020                                             

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
enrollment_year                                                                   
2009                                                         0.005615             
2010                                                         0.028077             
2011                                                         0.123540             
2012                                                         0.089847             
2013                                                         0.213387             
2014                                                         0.982704             
2015                                                         0.803010             
2016                                                         1.364555             
2017                                                         3.245732             
2018                                                         5.177448             
2019                                                         6.379155             
2020                                                         5.519991             

                                                \
presented_problems at risk of losing placement   
enrollment_year                                  
2009                                       NaN   
2010                                  0.016846   
2011                                  0.084232   
2012                                  0.056155   
2013                                  0.106694   
2014                                  0.219003   
2015                                  0.219003   
2016                                  0.320081   
2017                                  0.842318   
2018                                  1.050090   
2019                                  1.499326   
2020                                  1.196092   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
enrollment_year                                                            
2009                                                         0.005615      
2010                                                         0.011231      
2011                                                         0.039308      
2012                                                         0.056155      
2013                                                         0.095463      
2014                                                         0.179695      
2015                                                         0.185310      
2016                                                         0.297619      
2017                                                         0.920934      
2018                                                         1.684636      
2019                                                         2.201258      
2020                                                         1.869946      

                                                            \
presented_problems diagnosis and treatment plan assistance   
enrollment_year                                              
2009                                              0.005615   
2010                                              0.028077   
2011                                              0.078616   
2012                                              0.089847   
2013                                              0.117925   
2014                                              0.185310   
2015                                              0.179695   
2016                                              0.230234   
2017                                              0.887242   
2018                                              1.645328   
2019                                              1.982255   
2020                                             

In [31]:
demographics_problems_merged.enrollment_year.value_counts()

2019    4763
2020    4373
2018    3792
2017    2405
2016     950
2014     607
2015     501
2013     180
2011     114
2012      92
2010      26
2009       5
Name: enrollment_year, dtype: int64

In [32]:
count_year_proportion(filtered_combined_10)
count_year_distribution(filtered_combined_10)

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
enrollment_year                                                                   
2019                                                        10.237755             
2020                                                        12.771210             

                                                \
presented_problems at risk of losing placement   
enrollment_year                                  
2019                                  2.572431   
2020                                  2.767312   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
enrollment_year                                                            
2019                                                         3.403924      
2020                                                         4.326361      

                                                            \
presented_problems diagnosis and treatment plan assistance   
enrollment_year                                              
2019                                              3.209042   
2020                                              4.560218   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
enrollment_year                                                   
2019                              6.690919             3.105106   
2020                              9.094452             3.923607   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
enrollment_year                                                                 
2019                             5.768481        3.767702            1.649994   
2020                             8.133039        5.222814            2.052748   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
enrollment_year                                                                
2019                      0.012992          1.883851                 0.883461  
2020                      0.129921          2.520463                 1.312200

presented_problems_count  \
presented_problems aggression (physical, verbal, property destruction, threats)   
enrollment_year                                                                   
2019                                                              788             
2020                                                              983             

                                                \
presented_problems at risk of losing placement   
enrollment_year                                  
2019                                       198   
2020                                       213   

                                                                          \
presented_problems decrease in ability to participate in daily functions   
enrollment_year                                                            
2019                                                              262      
2020                                                              333      

                                                            \
presented_problems diagnosis and treatment plan assistance   
enrollment_year                                              
2019                                                   247   
2020                                                   351   

                                                                 \
presented_problems family needs assistance leaving unexpectedly   
enrollment_year                                                   
2019                                   515                  239   
2020                                   700                  302   

                                                                               \
presented_problems mental health symptoms self-injurious  sexualized behavior   
enrollment_year                                                                 
2019                                  444             290                 127   
2020                                  626             402                 158   

                                                                               
presented_problems suicidal action suicidal ideation transition from hospital  
enrollment_year                                                                
2019                             1               145                       68  
2020                            10               194                      101

In [33]:
filtered_combined_10.enrollment_year.value_counts()

2020    4373
2019    3324
Name: enrollment_year, dtype: int64

In [34]:
# emergency_crisis_services
# emergency_crisis_services.merge(demographics, left_on='Individual ID', right_on='Local ID',
#                                                                   how='inner', suffixes=('_emergency_services', '_demographics'), 
#                                                                   indicator = True).drop_duplicates()

In [35]:
living_situation_change.head()
living_situation_change.columns
living_situation_change.info()

Individual ID                    Region Current Living Situation  \
0          6942    Texas : El Paso County               Group home   
1       545329E     North Carolina : East               Group home   
2       724663E     North Carolina : East    Other: crisis respite   
3       717526C  North Carolina : Central               Group home   
4       351430W     North Carolina : West               Group home   

    New Living Situation                  Primary Reason for Move  \
0            Family home  Other: APS report came back unconfirmed   
1             Group home           Mental/behavioral health needs   
2  Other: crisis respite           Mental/behavioral health needs   
3             Group home           Mental/behavioral health needs   
4            Family home           Mental/behavioral health needs   

  Date of Move              Entered By  
0   2019-07-08         Angelica Garcia  
1   2018-05-01  David Jerome Lewis  II  
2   2019-06-01  David Jerome Lewis  II  
3   2019-07-31  Brittney Peters-Barnes  
4   2019-04-01          Aishah Bridges

Index(['Individual ID', 'Region', 'Current Living Situation',
       'New Living Situation', 'Primary Reason for Move', 'Date of Move',
       'Entered By'],
      dtype='object')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 601 entries, 0 to 600
Data columns (total 7 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   Individual ID             601 non-null    object        
 1   Region                    601 non-null    object        
 2   Current Living Situation  601 non-null    object        
 3   New Living Situation      601 non-null    object        
 4   Primary Reason for Move   601 non-null    object        
 5   Date of Move              593 non-null    datetime64[ns]
 6   Entered By                601 non-null    object        
dtypes: datetime64[ns](1), object(6)
memory usage: 33.0+ KB


In [36]:
# rename the column
living_situation_change.rename(columns={'Primary Reason for Move': 'move_reason'}, inplace=True)

# confirm that the column was renamed successfully
living_situation_change.columns

Index(['Individual ID', 'Region', 'Current Living Situation',
       'New Living Situation', 'move_reason', 'Date of Move', 'Entered By'],
      dtype='object')

In [41]:
living_situation_change_no_other = living_situation_change[~living_situation_change["move_reason"].str.contains('Other:')].copy()
print(f"The number of unique move reasons before dropping 'Other' is: {living_situation_change.move_reason.nunique()}")
print(f"The number of unique move reasons is: {living_situation_change_no_other.move_reason.nunique()}")

print(f"Retention Rate After Dropping 'Other': {round(100*living_situation_change_no_other.shape[0]/living_situation_change.shape[0], 2)}%")

The number of unique move reasons before dropping 'Other' is: 88
The number of unique move reasons is: 6
Retention Rate After Dropping 'Other': 83.86%


In [43]:
living_situation_change_no_other['move_date'] = pd.to_datetime(living_situation_change_no_other['Date of Move'])
print(f"Data type for the date variable: {living_situation_change_no_other['move_date'].dtype}")

# create a binary clarity column based on pre and post covid.
date_cutoff = pd.to_datetime("2020-03-01")
living_situation_change_no_other['is_post_covid'] = living_situation_change_no_other['move_date'] >= date_cutoff

# check the value counts to see how many of the inputs are post covid
living_situation_change_no_other['is_post_covid'].value_counts()


Data type for the date variable: datetime64[ns]


True     345
False    159
Name: is_post_covid, dtype: int64

In [44]:
# fucntion to cutoff date for living situation data
def compare_living_disparities(df: pd.DataFrame, month_bandwidth: int):
    pre_change_lower_bound = date_cutoff - relativedelta(months = month_bandwidth)
    post_change_upper_bound = date_cutoff + relativedelta(months = month_bandwidth)
    print(f"Pre Change Lower Bound: {pre_change_lower_bound}, Post Change Upper Bound: {post_change_upper_bound}")

    # Subset to data that is within the range of the pre and post covid data
    filtered_combined = df[(df['move_date'] >= pre_change_lower_bound) &
                                                    (df['move_date'] <= post_change_upper_bound)].copy()

    print(f"The shape for the filtered data frame is {filtered_combined.shape}")

    filtered_combined['is_post_covid'] = df['move_date'] >= date_cutoff

    return filtered_combined

filtered_living_12 = compare_living_disparities(living_situation_change_no_other, 12)
filtered_living_12['is_post_covid'].value_counts()

Pre Change Lower Bound: 2019-03-01 00:00:00, Post Change Upper Bound: 2021-03-01 00:00:00
The shape for the filtered data frame is (334, 9)


True     199
False    135
Name: is_post_covid, dtype: int64

In [47]:
def count_living_distribution(combined: pd.DataFrame):
    # see how the frequency of presented problem changed pre and post covid
    df = pd.DataFrame(combined.groupby('is_post_covid').agg({'move_reason' : 'value_counts'}).
                    rename(columns={'move_reason' : 'move_reason_count'}))

    # pivot the table to get the counts of presented problems pre and post covid
    return pd.pivot_table(df, index='is_post_covid', columns='move_reason')

count_living_distribution(filtered_living_12)

print()
# since the number entries for each presented problem before and post covid is different, 
# we need to normalize the value counts
def count_living_proportion(combined: pd.DataFrame):
    df = pd.DataFrame(combined.groupby('is_post_covid').agg({'move_reason' : 'value_counts'}).
                rename(columns={'move_reason' : 'move_reason_count'}))
    
    df['move_reason_count'] = (df['move_reason_count']/df['move_reason_count'].sum())*100

    return pd.pivot_table(df, index='is_post_covid', columns='move_reason')

count_living_proportion(filtered_living_12)

move_reason_count                                 \
move_reason   Aged out of current placement Change in family circumstances   
is_post_covid                                                                
False                                     3                              8   
True                                      1                             11   

                                                          \
move_reason   Individual/family preference Medical needs   
is_post_covid                                              
False                                   22             1   
True                                    61             3   

                                              \
move_reason   Mental/behavioral health needs   
is_post_covid                                  
False                                     90   
True                                     100   

                                                         
move_reason   Provider change (unrelated to individual)  
is_post_covid                                            
False                                                11  
True                                                 23

move_reason_count                                 \
move_reason   Aged out of current placement Change in family circumstances   
is_post_covid                                                                
False                              0.898204                       2.395210   
True                               0.299401                       3.293413   

                                                          \
move_reason   Individual/family preference Medical needs   
is_post_covid                                              
False                             6.586826      0.299401   
True                             18.263473      0.898204   

                                              \
move_reason   Mental/behavioral health needs   
is_post_covid                                  
False                              26.946108   
True                               29.940120   

                                                         
move_reason   Provider change (unrelated to individual)  
is_post_covid                                            
False                                          3.293413  
True                                           6.886228

In [50]:
in_patient_admissions.head()
in_patient_admissions.columns
in_patient_admissions.info()

Individual ID                    Region Date/Time of contact or admission  \
0       603802W     North Carolina : West               2015-05-01 00:00:00   
1       1187009    Texas : Tarrant County               2016-06-16 00:00:00   
2       1187009    Texas : Tarrant County               2016-08-31 00:00:00   
3       421925C  North Carolina : Central               2016-11-10 10:00:00   
4       1024977    Texas : Tarrant County               2016-06-01 18:00:00   

                                   Type of admission  \
0                         State psychiatric hospital   
1            Community mental health in-patient unit   
2            Community mental health in-patient unit   
3  State operated/private ICF/DD or other I/DD fa...   
4            Community mental health in-patient unit   

             Primary Reason for Contact or Admission  Length of Stay  \
0  Aggression (physical, verbal, property destruc...           67680   
1  Aggression (physical, verbal, property destruc...           10080   
2  Aggression (physical, verbal, property destruc...           10080   
3  Aggression (physical, verbal, property destruc...           54720   
4  Aggression (physical, verbal, property destruc...           28800   

  Facilitated by START Coordinator or START Team member  
0                                                 No     
1                                                 No     
2                                                 No     
3                                                Yes     
4                                                 No

Index(['Individual ID', 'Region', 'Date/Time of contact or admission',
       'Type of admission', 'Primary Reason for Contact or Admission',
       'Length of Stay',
       'Facilitated by START Coordinator or START Team member'],
      dtype='object')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2028 entries, 0 to 2027
Data columns (total 7 columns):
 #   Column                                                 Non-Null Count  Dtype         
---  ------                                                 --------------  -----         
 0   Individual ID                                          2028 non-null   object        
 1   Region                                                 2028 non-null   object        
 2   Date/Time of contact or admission                      2028 non-null   datetime64[ns]
 3   Type of admission                                      1979 non-null   object        
 4   Primary Reason for Contact or Admission                1979 non-null   object        
 5   Length of Stay                                         2028 non-null   int64         
 6   Facilitated by START Coordinator or START Team member  1951 non-null   object        
dtypes: datetime64[ns](1), int64(1), object(5)
memory usage: 111.0+ KB


In [54]:
# rename the column
in_patient_admissions.rename(columns={'Primary Reason for Contact or Admission': 'admission_reason'}, inplace=True)

# confirm that the column was renamed successfully
in_patient_admissions.columns

Index(['Individual ID', 'Region', 'Date/Time of contact or admission',
       'Type of admission', 'admission_reason', 'Length of Stay',
       'Facilitated by START Coordinator or START Team member'],
      dtype='object')

In [59]:
in_patient_admissions_no_other = in_patient_admissions[~in_patient_admissions["admission_reason"].str.contains('Other:', na=False)].copy()
print(f"The number of unique move reasons before dropping 'Other' is: {in_patient_admissions.admission_reason.nunique()}")
print(f"The number of unique move reasons is: {in_patient_admissions_no_other.admission_reason.nunique()}")

print(f"Retention Rate After Dropping 'Other': {round(100*in_patient_admissions_no_other.shape[0]/in_patient_admissions.shape[0], 2)}%")

The number of unique move reasons before dropping 'Other' is: 182
The number of unique move reasons is: 6
Retention Rate After Dropping 'Other': 89.0%


In [77]:
in_patient_admissions.admission_reason.unique()
in_patient_admissions_no_other.admission_reason.unique()
in_patient_admissions.shape
in_patient_admissions_no_other.shape

array(['Aggression (physical, verbal, property destruction, threats)',
       'Self-injurious', 'Suicidal ideation/behaviors',
       'Other: stating suicidal after she reported her old friend kissed her when she did not want to be kissed.',
       'Other: Psychosis and Aggitation', 'Other: mental health symptoms',
       'Other: psychosis, decreased verbal interactions',
       'Other: psychosis, aggression', 'Other: hallucinations',
       'Other: escalated impulisivity', 'Other: MH Symptoms',
       'Other: psychosis', 'Other: family abandonment',
       'Other: hallucinations/delusions/paranoia', nan,
       'Other: Aggression; 10/03/2015 MK',
       'Other: Aggression; 09/22/2015 MK',
       'Other: Medical and Psychiatric symptoms',
       'Other: Court Ordered Due to sexualized behavior',
       'Other: psychiatric decompensation/stabilization',
       'Other: Court ordered commitment due to criminal charges',
       'Other: sytmptomatic', 'Other: Evaluation/Possible Depression'

array(['Aggression (physical, verbal, property destruction, threats)',
       'Self-injurious', 'Suicidal ideation/behaviors', nan, 'Other',
       'Medical Concerns', 'Medication Evaluation'], dtype=object)

(2028, 7)

(1805, 9)

In [64]:
in_patient_admissions_no_other['admission_date'] = pd.to_datetime(in_patient_admissions_no_other['Date/Time of contact or admission'])
print(f"Data type for the date variable: {in_patient_admissions_no_other['admission_date'].dtype}")

# create a binary clarity column based on pre and post covid.
date_cutoff = pd.to_datetime("2020-03-01")
in_patient_admissions_no_other['is_post_covid'] = in_patient_admissions_no_other['admission_date'] >= date_cutoff

# check the value counts to see how many of the inputs are post covid
in_patient_admissions_no_other['is_post_covid'].value_counts()


Data type for the date variable: datetime64[ns]


False    1280
True      525
Name: is_post_covid, dtype: int64

In [69]:
def compare_admission_disparities(df: pd.DataFrame, month_bandwidth: int):
    pre_change_lower_bound = date_cutoff - relativedelta(months = month_bandwidth)
    post_change_upper_bound = date_cutoff + relativedelta(months = month_bandwidth)
    print(f"Pre Change Lower Bound: {pre_change_lower_bound}, Post Change Upper Bound: {post_change_upper_bound}")

    # Subset to data that is within the range of the pre and post covid data
    filtered_combined = df[(df['admission_date'] >= pre_change_lower_bound) &
                                                    (df['admission_date'] <= post_change_upper_bound)].copy()

    print(f"The shape for the filtered data frame is {filtered_combined.shape}")

    filtered_combined['is_post_covid'] = df['admission_date'] >= date_cutoff

    return filtered_combined

filtered_admission_10 = compare_admission_disparities(in_patient_admissions_no_other, 10)
filtered_admission_10['is_post_covid'].value_counts()

Pre Change Lower Bound: 2019-05-01 00:00:00, Post Change Upper Bound: 2021-01-01 00:00:00
The shape for the filtered data frame is (531, 9)


True     279
False    252
Name: is_post_covid, dtype: int64

In [73]:
def count_admissions_distribution(combined: pd.DataFrame):
    # see how the frequency of admission reasons changed pre and post covid
    df = pd.DataFrame(combined.groupby('is_post_covid').agg({'admission_reason' : 'value_counts'}).
                    rename(columns={'admission_reason' : 'move_reason_count'}))

    # pivot the table to get the counts of presented problems pre and post covid
    return pd.pivot_table(df, index='is_post_covid', columns='admission_reason')

count_admissions_distribution(filtered_admission_10)

# proportion for admission reasons comparing pre and post covid
def count_admissions_proportion(combined: pd.DataFrame):
    df = pd.DataFrame(combined.groupby('is_post_covid').agg({'admission_reason' : 'value_counts'}).
                rename(columns={'admission_reason' : 'admission_reason_count'}))
    
    df['admission_reason_count'] = (df['admission_reason_count']/df['admission_reason_count'].sum())*100

    return pd.pivot_table(df, index='is_post_covid', columns='admission_reason')

count_admissions_proportion(filtered_admission_10)

move_reason_count  \
admission_reason Aggression (physical, verbal, property destruction, threats)   
is_post_covid                                                                   
False                                                         155.0             
True                                                          158.0             

                                                                              \
admission_reason Medical Concerns Medication Evaluation Other Self-injurious   
is_post_covid                                                                  
False                         4.0                   1.0   1.0           22.0   
True                          6.0                   8.0   NaN           24.0   

                                              
admission_reason Suicidal ideation/behaviors  
is_post_covid                                 
False                                   67.0  
True                                    81.0

admission_reason_count  \
admission_reason Aggression (physical, verbal, property destruction, threats)   
is_post_covid                                                                   
False                                                     29.411765             
True                                                      29.981025             

                                                                   \
admission_reason Medical Concerns Medication Evaluation     Other   
is_post_covid                                                       
False                    0.759013              0.189753  0.189753   
True                     1.138520              1.518027       NaN   

                                                             
admission_reason Self-injurious Suicidal ideation/behaviors  
is_post_covid                                                
False                  4.174573                   12.713472  
True                   4.554080                   15.370019